README.md

```diff
@@ -22,20 +43,105 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+### Defining and Creating a Task
+
+To define a task, mix in the Task module:
+
+```ruby
+class FetchFile
+  include Task::Task
+
+  data_attr_reader :external_host
+  data_attr_reader :filename
+end
+```
+
+A task contains a set of data key-value pairs; The `data_attr_reader` helper provides accessors for various expected
+data fields, but is not required.
+
+To build a task, we invoke the build method:
+
+```ruby
+file_task = FetchFile.build(
+  id: 'file1',
+  task_list: 'datto.com',
+  external_host: 'datto.com',
+  filename: 'file1.txt')
+```
+
+Here, the task belongs to the 'datto.com' task list. The ID of the task should be unique across the task list (if one
```

Comment:
I'll clarify the documentation, but:
- if a duplicate ID/task_list pair is provided and the task is saved, the old one will be overwritten.
- Different backends would be independent.
- The ID needs to be unique within the task list, not globally.